### PR TITLE
Issues/#8142 add filters and columns in api list

### PR DESCRIPTION
**Pull request by Okhelifi**
**Issue**

https://github.com/gravitee-io/issues/issues/8142

**Description**

Add columns : state, context path, tags, quality, owner, visibility.

Screenshot 2022-08-30 at 12 39 06

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/issues-8142-filtered-api-list/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-shxqedfjfi.chromatic.com)

### CODE REVIEW

**Comment by mouligno** [comment]
@Okhelifi what's the url to play with it on https://apimnightlywebui24386.z6.web.core.windows.net/issues-8142-filtered-api-list/index.html ?

**Comment by Okhelifi** [comment]
> @Okhelifi what's the url to play with it on https://apimnightlywebui24386.z6.web.core.windows.net/issues-8142-filtered-api-list/index.html ?

to adapt with your environment if you are not running apim locally : 
http://localhost:3000/#!/environments/default/apis/list?page=1&size=10

File: gravitee-apim-console-webui/src/management/api/list/api-list.component.ts [hunk 1]
```diff
@@ -31,30 +33,62 @@ type ApisTableDS = { id: string; name: string }[];
...
   filters: GioTableWrapperFilters = {
     pagination: { index: 1, size: 10 },
     searchTerm: '',
   };
-  constructor(@Inject(UIRouterState) private readonly $state: StateService, private readonly apiService: ApiService) {}
+private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+  private filters$ = new BehaviorSubject<GioTableWrapperFilters>(this.filters);
+
+  constructor(
+    @Inject(UIRouterStateParams) private $stateParams,
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/list/api-list.component.ts]: Just a filling : 
This is not observable so use $ a the begin is confusing
Maybe `ajsStateParams` to be clear this comme from angularJs stuff 🤷‍♂️ 

(I think I'm the one who added the $ in the first 🙈 )

File: gravitee-apim-console-webui/src/management/api/list/api-list.component.spec.ts [hunk 2]
```diff
@@ -94,22 +95,40 @@ describe('ApisListComponent', () => {
...
+  it('should display out of sync api icon', fakeAsync(async () => {
+    const api = fakeApi();
+    const apis = [api];
+    await initComponent(apis);
+
+    const nameSort = await loader.getHarness(MatSortHeaderHarness.with({ selector: '#name' })).then((sortHarness) => sortHarness.host());
+    await nameSort.click();
+
+    expectSyncedApi(api.id, false);
+    expect(await loader.getHarness(MatIconHarness.with({ selector: '.states__api-is-not-synced' }))).toBeTruthy();
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/list/api-list.component.spec.ts]: WDYT to Check this synced icon is not displayer before expectSyncedApi ?

**Review by ThibaudAV — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/list/api-list.component.spec.ts (hunk 2 above)
Comment (Okhelifi) [reply gravitee-apim-console-webui/src/management/api/list/api-list.component.spec.ts]: Done

File: gravitee-apim-console-webui/src/management/api/list/api-list.component.ts (hunk 1 above)
Comment (Okhelifi) [reply gravitee-apim-console-webui/src/management/api/list/api-list.component.ts]: Done

**Pull request merged**